services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java

```diff
@@ -483,6 +484,87 @@ public Response updateNews(@Context HttpServletRequest request,
+    }
+    if (draftVisibility == null) {
+      return Response.status(Response.Status.BAD_REQUEST).build();
+    }
+    try {
+      newsService.setNewsDraftVisibility(newsId, draftVisibility);
+      return Response.ok().build();
+    } catch (Exception e) {
+      LOG.error("Error when updating the news  visibility" + newsId, e);
+      return Response.serverError().build();
+    }
+  }
+
+  @GET
+  @Path("isCurentUserRedactorOrAuthor/{spaceId}/{newsId}")
+  @RolesAllowed("users")
+  @ApiOperation(value = "Get a news illustration", httpMethod = "GET", response = Response.class, notes = "Check if current user is redactor on given space or author of news.")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Is redactor"),
+      @ApiResponse(code = 401, message = "User not authorized"),
+      @ApiResponse(code = 500, message = "Internal server error") })
+  public Response isCurentUserRedactorOrAuthor(@Context Request request,
+                                               @ApiParam(value = "space Id", required = true) @PathParam("spaceId") String spaceId,
+                                               @ApiParam(value = "news Id", required = true) @PathParam("newsId") String newsId) {
+    try {
+      if (StringUtils.isBlank(spaceId)) {
+        return Response.status(Response.Status.BAD_REQUEST).build();
+      }
+      Space space = spaceService.getSpaceById(spaceId);
+      if (space == null) {
+        return Response.status(Response.Status.NOT_FOUND).build();
+      }
+      JSONObject jsonObject = new JSONObject();
+      String authenticatedUser = ConversationState.getCurrent().getIdentity().getUserId();
+      boolean isRedactor = spaceService.isRedactor(space, authenticatedUser);
+      boolean isAuthor = newsService.isNewsAuthor(newsId, authenticatedUser);
+      jsonObject.put("isRedactor", isRedactor);
+      jsonObject.put("isAuthor", isAuthor);
+      return Response.ok().entity(jsonObject.toString()).build();
+    } catch (Exception e) {
+      LOG.error("Error when checking if the user is redactor of space ", e);
+      return Response.serverError().build();
+    }
+  }
+
+  @GET
+  @Path("getDraftVisibility/{newsId}")
+  @RolesAllowed("users")
+  @ApiOperation(value = "Get a news illustration", httpMethod = "GET", response = Response.class, notes = "Get the draft visibility")
+  @ApiResponses(value = { @ApiResponse(code = 200, message = "Draft visibility"),
+      @ApiResponse(code = 401, message = "User not authorized"),
+      @ApiResponse(code = 500, message = "Internal server error") })
+  public Response getDraftVisibility(@Context Request request,
+                                      @ApiParam(value = "news Id", required = true) @PathParam("newsId") String newsId) {
+    try {
+      if (StringUtils.isBlank(newsId)) {
+        return Response.status(Response.Status.BAD_REQUEST).build();
+      }
+      String drafttVisibility = newsService.getNewsDraftVisibility(newsId);
```

Comment:
security : where's the user identity and and permission check ?

Reply:
the PR is always under dev and not yet validated, the security checks will be ensured